packages/test/test-utils/src/containerRuntimeFactories.ts

```diff
@@ -0,0 +1,198 @@
+import type {
+	IFluidDataStoreFactory,
+	IFluidDataStoreRegistry,
+	IProvideFluidDataStoreRegistry,
+	NamedFluidDataStoreRegistryEntries,
+} from "@fluidframework/runtime-definitions/internal";
+import { RequestParser, RuntimeFactoryHelper } from "@fluidframework/runtime-utils/internal";
+
+const defaultDataStoreId = "default";
+
+async function getDefaultFluidObject(runtime: IContainerRuntime): Promise<FluidObject> {
+	const entryPoint = await runtime.getAliasedDataStoreEntryPoint("default");
+	if (entryPoint === undefined) {
+		throw new Error("default dataStore must exist");
+	}
+	return entryPoint.get();
+}
+
+/**
+ * {@link ContainerRuntimeFactoryWithDefaultDataStore} construction properties.
+ *
+ * @deprecated See notice on {@link ContainerRuntimeFactoryWithDefaultDataStore}.
+ */
+export interface ContainerRuntimeFactoryWithDefaultDataStoreProps {
+	readonly defaultFactory: IFluidDataStoreFactory;
+
+	/**
+	 * The data store registry for containers produced.
+	 */
+	readonly registryEntries: NamedFluidDataStoreRegistryEntries;
+
+	/**
+	 * The runtime options passed to the IContainerRuntime when instantiating it
+	 */
+	readonly runtimeOptions?: IContainerRuntimeOptions;
+
+	/**
+	 * Function that will initialize the entryPoint of the IContainerRuntime instances
+	 * created with this factory
+	 */
+	readonly provideEntryPoint?: (runtime: IContainerRuntime) => Promise<FluidObject>;
+}
+
+/**
+ * A ContainerRuntimeFactory that initializes Containers with a single default data store, which can be requested from
+ * the container with an empty URL.
+ *
+ * @deprecated
+ * Do not reference this type directly. It will be removed in the future.
+ * E.g. use {@link IRuntimeFactory} instead.
+ */
+export class ContainerRuntimeFactoryWithDefaultDataStore
+	extends RuntimeFactoryHelper
+	implements IProvideFluidDataStoreRegistry
+{
+	public static readonly defaultDataStoreId = defaultDataStoreId;
+
+	/**
+	 * {@inheritDoc @fluidframework/runtime-definitions#IProvideFluidDataStoreRegistry.IFluidDataStoreRegistry}
+	 */
+	public get IFluidDataStoreRegistry(): IFluidDataStoreRegistry {
+		return this.registry;
+	}
+
+	protected readonly defaultFactory: IFluidDataStoreFactory;
+
+	private readonly registry: IFluidDataStoreRegistry;
+
+	/**
+	 * {@inheritDoc ContainerRuntimeFactoryWithDefaultDataStoreProps.registryEntries}
+	 */
+	private readonly registryEntries: NamedFluidDataStoreRegistryEntries;
+
+	/**
+	 * {@inheritDoc ContainerRuntimeFactoryWithDefaultDataStoreProps.runtimeOptions}
+	 */
+	private readonly runtimeOptions?: IContainerRuntimeOptions;
+
+	// eslint-disable-next-line import/no-deprecated
+	private readonly requestHandlers: readonly RuntimeRequestHandler[];
+
+	/**
+	 * {@inheritDoc ContainerRuntimeFactoryWithDefaultDataStoreProps.provideEntryPoint}
+	 */
+	private readonly provideEntryPoint: (runtime: IContainerRuntime) => Promise<FluidObject>;
+
+	public constructor(props: ContainerRuntimeFactoryWithDefaultDataStoreProps) {
```

Comment:
I think it would be much more concise to make this just do
```suggestion
	public constructor(private readonly props: ContainerRuntimeFactoryWithDefaultDataStoreProps) {
```

and delete the other properties and their assignments. The `?? getDefaultFluidObject` can be inline into the one usage in preInitialize